pkg/ipam/ipam.go

```diff
@@ -122,6 +122,7 @@ func NewIPAM(nodeAddressing datapath.NodeAddressing, c Configuration, owner Owne
 			ipam.IPv4Allocator = newHostScopeAllocator(nodeAddressing.IPv4().AllocationCIDR().IPNet)
 		}
 	case option.IPAMCRD, option.IPAMENI, option.IPAMAzure:
+		// Do we need to split this one based on IPAMENI since it only supports IPv4.
```

Comment:
I don't think we need to do any check here. The daemon is the only piece that calls this function `NewIPAM()` and it does it after the `option.Config.Validate()` mentioned earlier.